### PR TITLE
Preview module

### PR DESCRIPTION
**Pull request by kuldev**

### CODE REVIEW

File: modules/ezcontent_preview/css/global-css.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.decoupled-content--preview {
+  border: none;
+  width: 100%;
+}
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/css/global-css.css]: Please leave a new line at EOF.

File: modules/ezcontent_preview/ezcontent_preview.info.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: EZContent Preview
+type: module
+description: Provide preview funtionality
+core: 8.x
+package: EZContent Preview
+configure: ezcontent.preview.config
+dependencies:
+  - access_unpublished
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.info.yml]: Please leave a new line at EOF.

File: modules/ezcontent_preview/ezcontent_preview.info.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: EZContent Preview
+type: module
+description: Provide preview funtionality
+core: 8.x
+package: EZContent Preview
+configure: ezcontent.preview.config
+dependencies:
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.info.yml:7]: Please add the module `access_unpublished` in composer.

File: modules/ezcontent_preview/ezcontent_preview.libraries.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+global:
+  version: 1.x
+  css:
+    theme:
+      css/global-css.css: {}
+  js:
+    js/global-js.js: {}
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.libraries.yml]: Please leave a new line at EOF.

File: modules/ezcontent_preview/ezcontent_preview.links.menu.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+ezcontent.preview.config:
+  title: 'EZContent preview'
+  route_name: ezcontent.preview.config
+  description: 'EZContent preview configuration'
+  parent: system.admin_config_system
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.links.menu.yml]: Please leave a new line at EOF.

File: modules/ezcontent_preview/ezcontent_preview.module [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            /*
+            $entity = $form_state->getFormObject();
+            if($entity->getEntity()->id()){
+                $form['actions']['preview']['#submit'][] = 'ezcontent_preview_form_submit';
+            } else {
+                $form['actions']['preview']['#access'] = FALSE;
+            }
+            */
+        }
+    }
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.module:21]: Please get the formatting correct.

File: modules/ezcontent_preview/ezcontent_preview.routing.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    _permission: 'EZContent preview config'
+    
+ezcontent.preview.view:
+  path: '/node/{node}/decoupled-preview'
+  defaults:
+    _controller: '\Drupal\ezcontent_preview\Controller\PreviewView::preview'
+    _title: 'EZContent view preview'
+  requirements:
+    _permission: 'EZContent view preview'
+    node: \d+
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.routing.yml]: Please leave a new line at EOF.

File: modules/ezcontent_preview/js/global-js.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+  Drupal.behaviors.ezcontent_preview = {
+    attach: function (context, settings) {
+      var iframe_obj = $(".decoupled-content--preview");
+      iframe_obj.height($(window).height());
+
+    }
+  };
+
+})(jQuery, Drupal);
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/js/global-js.js]: Please leave a new line at EOF.

File: modules/ezcontent_preview/src/Form/PreviewConfigForm.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    parent::submitForm($form, $form_state);
+    $this->config(static::SETTINGS)
+      ->set('ezcontent_preview_url', $form_state->getValue('ezcontent_preview_url'))
+      ->set('ezcontent_preview_token_expire_time', $form_state->getValue('ezcontent_preview_token_expire_time'))
+      ->save();
+  }
+
+
+
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Form/PreviewConfigForm.php]: Please remove additional lines.

**Review by vaibhavjain-in — CHANGES_REQUESTED** [review]

File: modules/ezcontent_preview/css/global-css.css (hunk 1 above)
Comment (kuldev) [reply modules/ezcontent_preview/css/global-css.css]: it should be there.

**Review by vaibhavjain-in — APPROVED** [review]

**Pull request merged**